packages/backend/src/accounting/psql/ledger-transfer/index.test.ts

```diff
@@ -513,4 +520,193 @@ describe('Ledger Transfer', (): void => {
       }
     )
   })
+
+  describe('getTransferForUpdate', (): void => {
+    test('gets transfer by transferRef', async (): Promise<void> => {
+      const transfer = await createLedgerTransfer(
+        {
+          creditAccountId: account.id,
+          debitAccountId: settlementAccount.id,
+          amount: 10n,
+          ledger: account.ledger
+        },
+        knex
+      )
+
+      const trx = await knex.transaction()
+
+      await expect(
+        getTransferForUpdate(serviceDeps, transfer.transferRef, trx)
+      ).resolves.toEqual(transfer)
+
+      await trx.commit()
+    })
+
+    test('returns undefined if no transfer found', async (): Promise<void> => {
+      const trx = await knex.transaction()
+
+      await expect(
+        getTransferForUpdate(serviceDeps, uuid(), trx)
+      ).resolves.toBeUndefined()
+
+      await trx.commit()
+    })
+
+    test('locks tranfser for update', async (): Promise<void> => {
+      const transfer = await createLedgerTransfer(
+        {
+          creditAccountId: account.id,
+          debitAccountId: settlementAccount.id,
+          amount: 10n,
+          ledger: account.ledger
+        },
+        knex
+      )
+
+      const trx = await knex.transaction()
+
+      await expect(
+        getTransferForUpdate(serviceDeps, transfer.transferRef, trx)
+      ).resolves.toEqual(transfer)
+
+      const timeoutOrTransfer = await Promise.race([
+        transfer.$query().patchAndFetch({ state: LedgerTransferState.PENDING }),
+        new Promise((reject) => setTimeout(() => reject('timeout'), 5000))
+      ])
```

Comment:
could definitely decrease the time, but I thought this was an ok method to test a lock